cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -58,15 +58,20 @@ namespace gpu {
  */
 __device__ __inline__ char const* seek_field_end(char const* begin,
                                                  char const* end,
-                                                 ParseOptions const& opts)
+                                                 ParseOptions const& opts,
+                                                 bool escape_char = false)
 {
   bool quotation = false;
   auto current   = begin;
   while (true) {
     // Use simple logic to ignore control chars between any quote seq
     // Handles nominal cases including doublequotes within quotes, but
-    // may not output exact failures as PANDAS for malformed fields
-    if (*current == opts.quotechar) {
+    // may not output exact failures as PANDAS for malformed fields.
+    // Check for instances such as "a2\"bc" and "\\" if `escape_char` is true.
+    if (*current == opts.quotechar and
+        (escape_char == false or
+         ((current == begin) or
+          ((current >= begin + 2) and (*(current - 1) != '\\' or *(current - 2) == '\\'))))) {
```

Comment:
Can we break this logic out in to it's own function, or some variables? I'm finding it difficult to follow the logic in its current form.

Reply:
why check the char at `current - 2` here?

Reply:
A scenario for "\\", where `\` is escaped.

Reply:
How does "\\" affect the quotation state?
We're only checking for unescaped quotes here.

Reply:
Yes, in case of "\\\\", the last quotation is isn't escaped, the backlash is esscaped.

Reply:
oooh. Wow, this IS hard to follow. I did not even notice that one of comparisons with `'\\'` is a negative.
big :+1: for @cwharris 's comment

Reply:
Let me know if the current version with the function is fine.